client/src/components/map/Map.tsx

```diff
@@ -1,106 +1,218 @@
+        .setLngLat([props.coordinates.lng, props.coordinates.lat])
+        .addTo(mapRef.current);
+
+      // Optionally, center the map on the new marker
+      mapRef.current.setCenter([props.coordinates.lng, props.coordinates.lat]);
+    }
+  }, [props.coordinates]);
+
+  useEffect(() => {
+    if (!mapRef.current) return;
+
+    const map = mapRef.current;
+
+    map.setZoom(props.zoom || defaultZoom);
+  }, [props.zoom]);
+
+  return (
+    <div
+      className={props.className}
+      ref={mapContainerRef}
+      id="map"
+      style={props.style}
+    ></div>
+  );
+};
 
 interface MapControlsProps {
-    // path type is temporary
-    onDone?: (path: number) => void
-    onCancel?: () => void
+  // path type is temporary
+  onDone?: (path: number) => void;
+  onCancel?: () => void;
 }
 
 const MapControls: React.FC<MapControlsProps> = (props) => {
-    return (
-        <Card
-            className="ring-transparent absolute top-0 sm:m-2 right-0 xsm:w-full sm:w-80 backdrop-blur bg-white/50"
+  return (
+    <Card className="ring-transparent absolute top-0 sm:m-2 right-0 xsm:w-full sm:w-80 backdrop-blur bg-white/50">
+      <TabGroup className="mt-1 flex justify-center">
+        <TabList variant="solid">
+          <Tab value="1">Point</Tab>
+          <Tab value="2">Area</Tab>
+          <Tab value="3">Whole Municipality</Tab>
+        </TabList>
+      </TabGroup>
+      <div className="mt-4 px-2 flex justify-between space-x-2">
+        <Button
+          size="xs"
+          variant="secondary"
+          icon={RiDeleteBinFill}
+          color="red"
+          className="flex-1"
+        >
+          Line
+        </Button>
+        <Button
+          size="xs"
+          variant="secondary"
+          icon={RiCloseLine}
+          onClick={props.onCancel}
+          className="flex-1"
         >
-            <TabGroup className="mt-1 flex justify-center">
-                <TabList variant="solid">
-                    <Tab value="1">Point</Tab>
-                    <Tab value="2">Area</Tab>
-                    <Tab value="3">Whole Municipality</Tab>
-                </TabList>
-            </TabGroup>
-            <div className="mt-4 px-2 flex justify-between space-x-2">
-                <Button size="xs" variant="secondary" icon={RiDeleteBinFill} color="red" className="flex-1">Line</Button>
-                <Button size="xs" variant="secondary" icon={RiCloseLine} onClick={props.onCancel} className="flex-1">Cancel</Button>
-                <Button size="xs" variant="primary" icon={RiCheckFill} onClick={() => {
-                    props.onDone!(1);
-                }} className="flex-1">Save</Button>
-            </div>
-        </Card>
-    )
+          Cancel
+        </Button>
+        <Button
+          size="xs"
+          variant="primary"
+          icon={RiCheckFill}
+          onClick={() => {
+            props.onDone!(1);
+          }}
+          className="flex-1"
+        >
+          Save
```

Comment:
when click on save reset the coordinates saved in the map.  Because when you reopen the map  for adding a new document two points become visible on the map at the same time